### DISCUSSION TITLE
The Future of Python 2 Support in This Project

### DISCUSSION THREAD

**Discussion by Mab879** [ComplianceAsCode/content#12046] [General]
Even though Python 2 support officially ended many years ago this project still supports building with Python 2 due to RHEL 7 and similar era operating systems still being supported and their default Python version of 2.7.

With the project dropping RHEL 7 support soon, see #12044, I was wondering if at that time we should also drop Python 2 support. It would make maintenance easier and allow use of newer Python features. Now don't get your hopes up too much, we looking supporting back to 3.6 (due to RHEL 8 and similar), but at least it won't be 2.7.

Please leave your thoughts here.

**Comment by Mab879** [comment]
@ComplianceAsCode/oracle-maintainers 
@ComplianceAsCode/suse-maintainers 
@ComplianceAsCode/ubuntu-maintainers

Do you still need Python 2.7 support to build your packages?

**Reply by dodys** [reply]
+1 from me on dropping py2 support

**Reply by Xeicker** [reply]
In oracle, we use python2 for OL7, in our case premium support ends until December 2024. So we would like to keep python2 support to build it until then

**Reply by Mab879** [reply]
@Xeicker Is there a container image of OL7 that could be used for testing Python 2.7 once the RHEL 7 testing is removed?

**Reply by Xeicker** [reply]
we have the dockerfile in here: https://github.com/ComplianceAsCode/content/blob/master/Dockerfiles/test_suite-ol7

**Reply by Mab879** [reply]
@Xeicker Do you still need Python 2 support in 2025?

**Reply by Xeicker** [reply]
Raised this topic with Oracle Linux team to decide. Will come with the answer shortly.

**Reply by Xeicker** [reply]
@Mab879  It is ok for us to remove python 2 support

**Reply by Mab879** [reply]
Thanks for the follow-up, this is great news.

**Comment by marcusburghardt** [comment]
We briefly discussed again this topic on https://github.com/ComplianceAsCode/content/pull/12816 . In that case, we would like to use type hints in SSG but this causes issues with Python 2.
